specfile/specfile.py

```diff
@@ -104,3 +113,66 @@ def tags(self, section: str = "package") -> Iterator[Tags]:
                 yield tags
             finally:
                 raw_section.data = tags.reassemble()
+
+    @contextlib.contextmanager
+    def changelog(self) -> Iterator[Optional[Changelog]]:
+        """
+        Context manager for accessing changelog.
+
+        Yields:
+            Spec file changelog as `Changelog` object or None if there is no %changelog section.
+        """
+        with self.sections() as sections:
+            try:
+                section = sections.changelog
+            except AttributeError:
+                yield None
+            else:
+                changelog = Changelog.parse(section)
+                try:
+                    yield changelog
+                finally:
+                    section.data = changelog.reassemble()
+
+    def add_changelog_entry(
+        self,
+        entry: str,
+        author: Optional[str] = None,
+        email: Optional[str] = None,
+        date: Optional[datetime.date] = None,
+    ) -> None:
+        """
+        Adds a new %changelog entry. Does nothing if there is no %changelog section.
+
+        If not specified, author and e-mail will be determined using rpmdev-packager, if available.
+        Date, if not set, will be set to today.
+
+        Args:
+            entry: Entry text.
+            author: Author of the entry.
+            email: E-mail of the author.
+            date: Date of the entry.
+        """
+
+        def get_author():
+            if author is not None:
+                return author + (f" <{email}>" if email else "")
+            try:
+                result = subprocess.check_output("rpmdev-packager").decode().strip()
+            except subprocess.CalledProcessError as e:
+                raise SpecfileException("Failed to auto-detect author") from e
+            else:
+                return result
+
+        with self.changelog() as changelog:
+            if not changelog:
+                return
+            evr = self.expand(
+                "%{?epoch:%{epoch}:}%{version}-%{release}", extra_macros=[("dist", "")]
+            )
+            changelog.insert(
+                0,
+                ChangelogEntry(
+                    date or datetime.date.today(), get_author(), evr, [f"- {entry}"]
```

Comment:
This seems to assume that there is only a single "entry" in one changelog. But what if we wanted to create something like:
```
<header>
- something
- something else
```

Reply:
Thinking about it, I'd probably find more use cases for adding entries with mutiple lines myself. I'm going to change it.